src/lib.rs

```diff
@@ -227,41 +229,89 @@ pub mod pallet {
 			Ok(Default::default())
 		}
 
-		/// First claim for collecting vested tokens. This one is free
-		#[pallet::weight((0, DispatchClass::Normal, Pays::No))]
-		pub fn my_first_claim(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
+		/// Collect whatever portion of your reward are currently vested. The first time each
+		/// contributor calls this function pays no fees
+		#[pallet::weight(0)]
+		pub fn show_me_the_money(origin: OriginFor<T>) -> DispatchResultWithPostInfo {
```

Comment:
I'm not sure this is going to work. Once we have done the benchmarking we will replace this `weight(0)` with a proper weight. Transactions that call this function only pass the preliminary validity checks if they can afford to pay for their annotated weight.

It's true that you can, in the end, charge less weight than was included in the annotation, but the user still has to be able to pay for it up front.

So if the goal is just that users won't incur a fee for their first claim, it is possible. But they still have to have enough tokens to cover what would have been the fee. So if the goal is that a user who has 0 tokens can get her first token by making a free claim, that won't work with a single extrinsic.

Reply:
Interesting, did not know that. @crystalin What do you think about this?

Reply:
I can try it out in any case, will get back with the results

Reply:
Finally the free claim has been discarded